python/ray/tune/ray_trial_executor.py

```diff
@@ -115,7 +112,13 @@ def logger_creator(config):
 
         # Logging for trials is handled centrally by TrialRunner, so
         # configure the remote runner to use a noop-logger.
-        return cls.remote(config=trial.config, logger_creator=logger_creator)
+        remote_runner = cls.remote(trial.config, logger_creator=logger_creator)
+        tune_config = {
+            "keep_checkpoints_num": trial.keep_checkpoints_num,
+            "checkpoint_score_attr": trial.checkpoint_score_attr,
+        }
+        remote_runner._tune_setup.remote(tune_config)
```

Comment:
Also, why wouldn't it make sense to put all of the checkpoint-management things in a separate class?

Reply:
Well so we'd have to then pull out all of `save`, `save_to_object`, `restore` and `restore_from_object`. There's no reason we can't do that, it just seemed like a lot for this PR but I can make those changes.